src/transmission_manager.rs

```diff
@@ -0,0 +1,293 @@
+            InfectiousStatus::Recovered,
+        );
+    }
+    Ok(())
+}
+
+fn get_next_infection_time_unif(context: &mut Context, last_infection_time_unif: f64) -> f64 {
+    // this is NOT order statistics
+    // we are just playing around to make a point
+    // we will update the math to be correct later
+    context.sample_range(TransmissionRng, last_infection_time_unif..1.0)
+}
+
+fn get_next_infection_time_from_gi(
+    _context: &mut Context,
+    gi: f64,
+    next_infection_time_unif: f64,
+) -> f64 {
+    // this will be properly stored later
+    //let gi = context.get_data_container(generation_interval).unwrap();
+    // the math here is wrong as well -- we are not implementing order statistics
+    // there is no guarantee that these infection attempts are sequential draws from the GI distribution
+    // this will be fixed with real math later
+    Exp::new(1.0 / gi)
+        .unwrap()
+        .inverse_cdf(next_infection_time_unif)
+}
+
+fn schedule_next_infection_attempt(
+    context: &mut Context,
+    transmitee_id: PersonId,
+    gi: f64,
+    num_infection_attempts_remaining: f64,
+    last_infection_time_unif: f64,
+) {
+    // get next infection attempt time
+    let next_infection_time_unif = get_next_infection_time_unif(context, last_infection_time_unif);
+    let next_infection_time_gi =
+        get_next_infection_time_from_gi(context, gi, next_infection_time_unif);
+    // schedule the infection attempt: this grabs a contact at the time of the infection event
+    // to make sure the contacts are based on who is alive at that time and evaluates whether the
+    // transmission event is successful
+    context.add_plan(
+        context.get_current_time() + next_infection_time_gi,
+        move |context| {
+            infection_attempt(
+                context,
+                transmitee_id,
+                gi,
+                num_infection_attempts_remaining,
+                next_infection_time_unif,
+            )
+            .expect("Error finding contact in infection attempt");
+        },
+    );
+}
+
+fn handle_infectious_status_change(
+    context: &mut Context,
+    event: PersonPropertyChangeEvent<InfectiousStatusType>,
+    r_0: f64,
+    gi: f64,
+) {
+    // is the person going from S --> I?
+    // we don't care about the other cases here
+    if matches!(event.previous, InfectiousStatus::Susceptible) {
+        // get the number of infection attempts this person will have
+        // ok to use unwrap here because we pass inputs (r_0) through a validator
+        let num_infection_attempts =
+            context.sample_distr(TransmissionRng, Poisson::new(r_0).unwrap());
+        // start scheduling infection attempt events for this person
+        // people who have num_infection_attempts = 0 are still passed through this logic so that
+        // they are set to recovered
+        schedule_next_infection_attempt(context, event.person_id, gi, num_infection_attempts, 0.0);
```

Comment:
Why are you carrying `gi` around like this rather than either (1) storing it witht he person or (2) getting it from parameters each time?